packages/now-cli/src/commands/env/add.ts

```diff
@@ -0,0 +1,153 @@
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import { NowContext, ProjectEnvTarget } from '../../types';
+import { Output } from '../../util/output';
+import Client from '../../util/client';
+import stamp from '../../util/output/stamp';
+import { getLinkedProject } from '../../util/projects/link';
+import addEnvRecord from '../../util/env/add-env-record';
+import {
+  isValidEnvTarget,
+  validEnvTargets,
+  getEnvTargetChoices,
+} from '../../util/env/env-target';
+import readStandardInput from '../../util/input/read-standard-input';
+import cmd from '../../util/output/cmd';
+import withSpinner from '../../util/with-spinner';
+import { emoji, prependEmoji } from '../../util/emoji';
+
+type Options = {
+  '--debug': boolean;
+};
+
+export default async function add(
+  ctx: NowContext,
+  opts: Options,
+  args: string[],
+  output: Output
+) {
+  const {
+    authConfig: { token },
+    config,
+  } = ctx;
+  const { currentTeam } = config;
+  const { apiUrl } = ctx;
+  const debug = opts['--debug'];
+  const client = new Client({ apiUrl, token, currentTeam, debug });
+  const link = await getLinkedProject(output, client);
+
+  if (link.status === 'error') {
+    return link.exitCode;
+  } else if (link.status === 'not_linked') {
+    output.print(
+      `${chalk.red(
+        'Error!'
+      )} Your codebase isn’t linked to a project on ZEIT Now. Run ${cmd(
+        'now'
+      )} to link it.\n`
+    );
+    return 1;
+  } else {
+    const { org, project } = link;
+    if (args.length > 2) {
+      output.error(
+        `Invalid number of arguments. See: ${chalk.cyan(
+          '`now env --help`'
+        )} for usage.`
+      );
+      return 1;
+    }
+
+    let envValue = await readStandardInput();
+    const addStamp = stamp();
+    let [envName, envTarget] = args;
+
+    let envTargets: ProjectEnvTarget[] = [];
+    if (envTarget) {
+      if (!isValidEnvTarget(envTarget)) {
+        output.error(
+          `The environment ${cmd(envTarget)} is not valid.\n` +
+            `Please use one of the following: <${validEnvTargets().join(
+              ' | '
+            )}>.`
+        );
+        return 1;
+      }
+      envTargets.push(envTarget);
+    }
+
+    while (!envName) {
+      const { inputName } = await inquirer.prompt({
+        type: 'input',
+        name: 'inputName',
+        message: `What’s the name of the variable?`,
+      });
+
+      if (!inputName) {
+        output.error(`Name cannot be empty`);
+        continue;
+      }
+
+      envName = inputName;
+    }
+
+    while (!envValue) {
+      const { inputValue } = await inquirer.prompt({
+        type: 'password',
+        name: 'inputValue',
+        message: `What’s the value of ${envName}?`,
+        /*transformer: (txt: string) => {
+          return `${chalk.gray(chalk.italic('*'.repeat(txt.length)))}`;
+        },*/
```

Comment:
Should we remove that?

Reply:
Oh thanks for pointing this out! I was leaving it here to get feedback :smart:
The default `password` prompt is similar to `sudo` in that what you type is hidden.
The transformer here was an alternative so that it shows up as asterisks instead of being hidden.

I'm okay with either solution because I would imagine that most secrets will use stdin like `cat secret.txt | now env add` because nobody wants to type.

What are your thoughts?

Reply:
I don't feel strongly. What's the most "steardy" solution, in your opinion? I'd go for that.

Reply:
Ok I don't feel strongly either way so I deleted the comment 👍